pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -321,7 +339,7 @@ int cmp_npy_datetimestruct(const npy_datetimestruct *a,
  * Returns -1 on error, 0 on success, and 1 (with no error set)
  * if obj doesn't have the needed date or datetime attributes.
  */
-int convert_pydatetime_to_datetimestruct(PyDateTime_Date *dtobj,
+int convert_pydatetime_to_datetimestruct(PyDateTime_DateTime *dtobj,
```

Comment:
Types are pretty loosely checked in this module; changed these to a more exact date time for clarity